packages/node/src/utils/promise-utils.ts

```diff
@@ -87,6 +68,28 @@ export interface ContinuousRetryOptions {
   delay?: number;
 }
 
+// A native implementation of the following function might look like:
+//
+//   function promiseTimeout<T>(ms: number, promise: Promise<T>): Promise<T> {
+//     const timeout = new Promise((_res, reject) => {
+//       setTimeout(() => {
+//         reject(new Error(`Timed out in ${ms} ms.`));
+//       }, ms);
+//     });
+//     return Promise.race([promise, timeout]);
+//   }
+//
+// The problem with this is that that the slow promise still runs until it resolves.
+// This means that the serverless function will not exit until the entire timeout
+// duration has been reached which is a problem.
+export function promiseTimeout<T>(ms: number, promise: Promise<T>): Promise<T> {
+  return Bluebird.resolve(promise).timeout(ms);
+}
+
+export function wait(ms: number) {
+  return new Promise((r) => setTimeout(r, ms));
+}
+
 export function retryOnTimeout<T>(maxTimeoutMs: number, operation: () => Promise<T>, options?: ContinuousRetryOptions) {
```

Comment:
Is this function still needed?

Reply:
Yeah, but it's only used when executing an API call at the moment